✨ feat(event): Listar eventos na tela home

joellacerda opened the pull request:
- Implemented `HomeViewModel` to fetch and distribute events.
- Updated `HomeViewController` to use the `HomeViewModel` and display the next event.

joellacerda committed:
🐛 fix(event): Fix date handling and display

- Modified the `Event` model to store dates as `Date` objects instead of `String`.
- Updated the `EventDTO` model to use `Date` type for the `date` property.
- Adjusted the `CreateEventViewController` to pass `Date` objects to the `Event` model.
- Modified `EventCardTableViewCell` to properly format and display the event date.
- Fixed `EditEventDetailsView` and `EditEventViewModel` to handle date updates correctly.
- Changed `EventDetailsView` to format the date using `DateFormatter` and localized strings.
- Added `EventsListView` and `EventsListViewController` to display a list of events.
- Implemented `HomeViewModel` to fetch and distribute events.
- Updated `HomeViewController` to use the `HomeViewModel` and display the next event.

joellacerda committed:
🔨 chore(home): Remove EventsList and update HomeView constraints

- Removed EventsList related files: EventsListView, EventsListViewController, and EventsListViewModel.
- Adjusted constraints in HomeView to accommodate the removal of the events list.
- Added tableViewDelegate to HomeViewController to manage tableView selection.
- Set a fixed height to eventCardView on HomeView

joellacerda commented:
@Davidap041 pra fazer a listagem dos eventos, eu tive que alterar o atributo `date` de Event para o tipo Date ao invés de String. Assim ele pega os eventos dos mais próximos para os mais antigos. Outra coisa, acho que tá meio redundante ter o próximo evento tanto na Home quanto nos Meus Eventos. O que você achar de remover isso dessa tela e deixar só os destaques?

Davidap041 commented:
Deixa só itens em destaques e não altera a model, só converte para string, tem extensão só pra fazer essa conversão de string para date e date para string .
Tem lá na criar eventos como exemplo

Davidap041 commented:
O event DTO também não precisa ser mudado

Davidap041 commented:
Altera só a home se for para corrigir algo na create cria uma branch só pra isso por favor

joellacerda committed:
🐛 fix(event): Change event date type from Date to String

- Modified the Event model, DTO, and related views to use String for the date instead of Date to handle date formatting more consistently.
- Updated the CreateEventViewController to convert the selected date to a String before saving.

joellacerda commented:
Bugs a resolver:

<img width="518" alt="image" src="https://github.com/user-attachments/assets/c0ee4747-6de9-473f-ad77-7b1fc5e76ef8" />

joellacerda committed:
✨ feat(event): Improve date formatting and remove unused code

- Updated date formatting in `CreateEventViewController` to use `date.toString()` for consistency and readability.
- Removed commented-out code blocks in `CreateEventView` and `EventDetailsView` to clean up the codebase.

joellacerda committed:
🔨 chore(CreateEvent): Remove commented-out code and fix date formatting in mock data
- Removed the commented-out `setupCategoryPicker` function in `CreateEventView.swift`.
- Fixed a typo in the `createMockDate` function definition in `Events+MockData.swift`.

joellacerda committed:
🎨 style(CreateEvent): Format commented-out code and update date label

- Reformatted the commented-out `setupCategoryPicker` function in `CreateEventView.swift` for better readability.
- Updated the way the date label is updated in `CreateEventViewController.swift` to improve code consistency.

joellacerda committed:
🎨 style(CreateEventView): Minor code formatting changes in CreateEventView

- Added a missing space after the `//` in a comment to improve readability.
